daemon/snap.go

```diff
@@ -120,10 +118,18 @@ func allLocalSnapInfos(st *state.State, all bool, wanted map[string]bool) ([]abo
 		}
 		var aboutThis []aboutSnap
 		var info *snap.Info
-		var publisher *snap.StoreAccount
 		var err error
 		if all {
+			var publisher snap.StoreAccount
+			var gotPublisher bool
 			for _, seq := range snapst.Sequence {
+				if !gotPublisher {
+					gotPublisher = true
+					publisher, err = publisherAccount(st, seq.SnapID)
```

Comment:
Is it posible to have the publisher change across revisions?

Reply:
hmmm. That's the same question as “can a snap ID change between revisions”. And the answer is yes: you can have a local revision (snap ID would be `""`, publisher unknown) and then a store revision with a given snap ID and publisher.

Reply:
I've made it grab the publisher for every revision, and added a test for the simplest way this can occur (the local-vs-from-store I described). Note that it is also in theory possible for it to happen via some stranger mechanism, but it hasn't ever happened yet :-)

The simplest way of seeing this in action is having a from-the-store snap, installing something from a snap with the same name using `--dangerous`, and then doing `snap list --all`: you should see, e.g.,
```
$ snap list icdiff --all
Name    Version  Rev  Tracking  Publisher  Notes
icdiff  1.9.1-1  1    stable    chipaca    disabled
icdiff  1.9.4-1  x1   stable    -          -
```
if you see any publisher there for the local revision, it's wrong :-)